riscv/plic.cc

```diff
@@ -313,8 +316,16 @@ bool plic_t::load(reg_t addr, size_t len, uint8_t* bytes)
       return false;
   }
 
-  if (PRIORITY_BASE <= addr && addr < ENABLE_BASE) {
+  if (PRIORITY_BASE <= addr && addr < PENDING_BASE) {
     ret = priority_read(addr, &val);
+  } else if (PENDING_BASE <= addr && addr < ENABLE_BASE) {
+    uint32_t id_word = (addr - PENDING_BASE) >> 2;
+    if (id_word < num_ids_word) {
+      val = level[id_word];
```

Comment:
This is not correct. The pending register should be an ORing of pending state in each context

```
val = 0;
for (i = 0; i < contexts.size(); i++) {
    val |= contexts.pending[id_word];
}
```

Reply:
Done.